readme.adoc

```diff
@@ -63,15 +69,83 @@ algorithm.
 
 === List
 
-Allows for listing all Seed compliant images residing on the local system.
+Simple command to list the local seed compliant images.  It can be run with the following command:
+
+----
+seed list
+----
 
 === Search
 
-Allows for discovery of Seed compliant images hosted within a Docker registry.
+Allows for discovery of Seed compliant images hosted within a Docker registry. The 'seed search' command will search
+a given registry for repositories (images and all their versions) that end in "-seed".  Here is an example of searching quay.io:
+
+----
+seed search -r https://quay.io
+----
+
+If no registry url is supplied, the search command uses docker hub, which requires a organization (or user) to be specified:
+
+----
+seed search -o geoint
+----
+
+If a registry is private, a username and password can be specified with -u and -p options:
+
+----
+seed search -r http://localhost:5000 -u testuser -p testpassword
+----
 
 === Publish
 
-Allows for publish of Seed compliant images to a remote Docker registry.
+Provides a convenient way for algorithm developers to push a seed image to a registry.  This command will tag a seed
+image appropriately and push it to the specified registry.  If successful, it will then remove the local image as well.
+Here is an example of publishing the extractor image to the geoint organization on docker hub:
+
+----
+seed publish -in extractor-0.1.0-seed:0.1.0 -r docker.io -o geoint
+----
+
+Publishing will check if an image with the same name and tag exists in the registry and will fail if one is found unless
+either the force flag (-f) is set or a deconflict tag is specified to increase a version number.  A common use case for
+seed algorithm developers is to publish new versions of their image and this can be done by specifying one of the job or
+package version flags.  Here is an example of updating the extractor image to 1.0.0:
+
+----
+seed publish -in extractor-0.1.0-seed:0.1.0 -r docker.io -o geoint -A -P
+----
+
+This will rebuild the extractor image with the appropriate name & label and publish the image extractor-1.0.0-seed:1.0.0
+to docker hub.  Finally, if the registry is private, a username and password can be specified with -u and -p options:
+
+----
+seed publish -in extractor-0.1.0-seed:0.1.0 -r localhost:5000 -u testuser -p testpassword
+----
+
+=== Validate
+
+The Validate command will validate a seed json file against the seed schema.  This is also done as part of the build and
```

Comment:
Minor nitpick, but we should capitalize any reference to Seed other than when it is to the `seed.manifest.json` or similar files.